### PR TITLE
Run docker-distribution in a container

### PR DESCRIPTION
**Pull request by hardys**
There is no RHEL8 package for this AFAICS so lets run it
in a container which should work on both centos7 and el8 hosts.

### CODE REVIEW

File: 01_install_requirements.sh [hunk 1]
```diff
@@ -28,7 +27,9 @@ ANSIBLE_FORCE_COLOR=true ansible-playbook \
 popd
 
 # needed if we are using locally built images
-sudo systemctl start docker-distribution
+if ! sudo podman ps | grep -q registry; then
```
[the remark this answers is not in the document]
Comment (hardys) [reply 01_install_requirements.sh]: Ah good point, I'll check if the systemd service exists and stop/disable it if so.

File: 01_install_requirements.sh [hunk 2]
```diff
@@ -28,7 +27,9 @@ ANSIBLE_FORCE_COLOR=true ansible-playbook \
...
+  sudo podman run -d -p 5000:5000 --name registry docker.io/registry:latest
```
[the remark this answers is not in the document]
Comment (hardys) [reply 01_install_requirements.sh]: Ah, yeah I wasn't sure if we'd prefer not to delete it so images are cached between runs, maybe we should just restart it if it's not running?

[the remark this answers is not in the document]
Comment (derekhiggins) [reply 01_install_requirements.sh]: Yup restart would do, it should keep the cache

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1211/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1220/

File: 01_install_requirements.sh [hunk 3]
```diff
@@ -28,7 +27,16 @@ ANSIBLE_FORCE_COLOR=true ansible-playbook \
 popd
 
 # needed if we are using locally built images
-sudo systemctl start docker-distribution
+# We stop any systemd service so we can run in a container, since
+# there's no RPM/systemd version available for RHEL8
+if sudo systemctl is-active docker-distribution.service; then
+  sudo systemctl disable --now docker-distribution.service
+fi
+reg_state=$(sudo podman inspect registry --format  "{{.State.Status}}")
```
Comment (stbenjam) [inline 01_install_requirements.sh]: This is failing in CI:

```
Oct 11 14:56:11 |  + sudo systemctl is-active docker-distribution.service
Oct 11 14:56:11 |  unknown
Oct 11 14:56:11 |  ++ sudo podman inspect registry --format '{{.State.Status}}'
Oct 11 14:56:12 |  Error: error getting image "registry": unable to find a name and tag match for registry in repotags: no such image
```

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1232/

**Comment by stbenjam** [comment]
Failure is unrelated and has to do with something in BMO/CAPBM, merging this so I can stop applying this patch locally.

**Pull request merged**